autonomi/python/examples/autonomi_scratchpad.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python3
+"""
+Autonomi Scratchpad Example
+
+This example demonstrates how to use the scratchpad functionality in Autonomi,
+including how the counter is automatically managed during updates.
+
+Scratchpads are mutable, encrypted data containers on the Autonomi network.
+Each scratchpad is tied to a specific public key and can be updated by the owner.
+"""
+
+from autonomi_client import (
+    Client, SecretKey, Wallet, PaymentOption, Network, Bytes,
+    Scratchpad, ScratchpadAddress
+)
+import asyncio
+import time
+
+async def main():
+    # Initialize client
+    print("Connecting to Autonomi network...")
+    client = await Client.init_local()  # For testnet use Client.init()
+    
+    # Create wallet for payment
+    print("Setting up wallet...")
+    network = Network(True)  # Testnet
+    private_key = "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80"
+    wallet = Wallet.new_from_private_key(network, private_key)
+    payment = PaymentOption.wallet(wallet)
+    
+    # Create secret key for the scratchpad (this represents your identity)
+    key = SecretKey.random()
+    public_key = key.public_key()
+    print(f"Generated key with public key: {public_key.hex}")
+    
+    # 1. Create a new scratchpad
+    content_type = 42  # Application-specific type identifier
+    initial_data = b"My first scratchpad on Autonomi network!"
+    
+    # Check the cost before creating
+    cost_estimate = await client.scratchpad_cost(public_key)
+    print(f"Estimated cost for scratchpad: {cost_estimate}")
+    
+    print("Creating scratchpad...")
+    cost, addr = await client.scratchpad_create(key, content_type, initial_data, payment)
+    print(f"Scratchpad created at {addr.hex}, cost: {cost}")
+    
+    # Wait for network replication
+    print("Waiting for network replication...")
+    time.sleep(2)
+    
+    # 2. Retrieve the scratchpad
+    print("Retrieving scratchpad...")
+    scratchpad = await client.scratchpad_get(addr)
+    
+    # Display scratchpad information
+    print(f"Scratchpad owner: {scratchpad.address().owner().hex}")
+    print(f"Scratchpad content type: {scratchpad.data_encoding()}")
+    print(f"Scratchpad counter: {scratchpad.counter()}")  # Should be 0 for new scratchpad
```

Comment:
These two methods, `data_encoding` and `counter` do not exist. We can add them in this PR.